core/src/main/java/com/netflix/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -69,30 +73,33 @@ public TableMetadata current() {
   @Override
   public TableMetadata refresh() {
     int ver = version != null ? version : readVersionHint();
-    Path metadataFile = metadataFile(ver);
-    FileSystem fs = getFS(metadataFile, conf);
+    InputFile metadataFile = io().readMetadataFile(metadataFileName(ver));
+    Path metadataFilePath = new Path(metadataFile.location());
+    FileSystem fs = getFS(metadataFilePath, conf);
     try {
       // don't check if the file exists if version is non-null because it was already checked
-      if (version == null && !fs.exists(metadataFile)) {
+      if (version == null && !fs.exists(metadataFilePath)) {
```

Comment:
Yeah I also understand not wanting to add `exists`. I'm also hesitant because we don't want to blow up these APIs to become a full-fledged file system. Where is the boundary with regards to what File System-like operations we will require implementations to provide?

Reply:
We want to keep it as light-weight as possible to make it easy to avoid using the Hadoop API in the future. There's no reason to force downstream apps to include it. After that goal is met, then it's really a matter of trying to do as much as we can cleanly through the API. I think that adding exists would be okay. Then the only non-standard part of the Hadoop implementation would be the rename call.